src/components/DataProvider/index.js

```diff
@@ -197,22 +199,23 @@ export default class DataProvider extends Component {
     clearInterval(this.fetchInterval);
   }
 
-  static getXSubDomain = (xDomain, xSubDomain) => {
+  static getXSubDomain = (xDomain, xSubDomain, limitXSubDomain) => {
     if (!xSubDomain) {
       return xDomain;
     }
+    const newXSubDomain = limitXSubDomain(xSubDomain);
```

Comment:
Won't this fail if `limitXSubDomain` is not specified? Could we have it default to `subDomain => subDomain`?